File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    input = String.trim(input)
    cond do
      input == "" -> "Fine. Be that way!"
      String.last(input) == "?" -> "Sure."
      String.upcase(input) == input && String.downcase(input) != input -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as s]))

(defn shouting? [statement]
  (and (re-find #"[a-zA-Z]" statement) (= (s/upper-case statement) statement)))

(defn response-for [statement]
  (cond
    (s/blank? statement) "Fine. Be that way!"
    (shouting? statement) "Whoa, chill out!"
    (s/ends-with? statement "?") "Sure."
    :else "Whatever."))

